src/shared/filesystemUtilities.ts

```diff
@@ -220,3 +221,21 @@ export async function cloud9Findfile(dir: string, fileName: string): Promise<vsc
     }
     return []
 }
+/**
+ * @returns  A string path to the last locally stored download location. If none, returns the users 'Downloads' directory path.
+ */
+export function getDefaultDownloadPath(): string {
+    const lastUsedPath = globals.context.globalState.get('aws.downloadPath')
+    if (lastUsedPath) {
+        if (typeof lastUsedPath === 'string') {
+            return lastUsedPath
+        } else {
+            getLogger().verbose("Expected stored 'aws.downloadPath' to return a string")
+        }
+    }
+    return downloadsDir()
+}
+
+export function setDefaultDownloadPath(downloadPath: string) {
+    globals.context.globalState.update('aws.downloadPath', path.dirname(downloadPath))
```

Comment:
This assumes that `downloadPath` points to a file. That isn't clear from the function interface or docstring. Worth adding a isdirectory() check.